checker-qual/src/main/java/org/checkerframework/checker/mustcall/qual/CreatesObligation.java

```diff
@@ -0,0 +1,95 @@
+package org.checkerframework.checker.mustcall.qual;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.checkerframework.framework.qual.InheritedAnnotation;
+import org.checkerframework.framework.qual.JavaExpression;
+
+/**
+ * Indicates that the method resets the must-call type on the target (the value argument) to the
+ * greatest lower bound of its current must-call type and its declared must-call type. This
```

Comment:
I meant LUB, thanks for noticing. I've fixed the documentation.

As for the rest of your question: the LUB isn't always the declared type (though it usually is). For example, a programmer might annotate a method's return type as `@MustCall("foo") Object`. If a variable derived from calling that method becomes the target of an `@CreateObligation` annotation, then it retains the type `@MustCall("foo")`, because the LUB of that and its declared type of `Object` (which is implicitly `@MustCall({})`) is `@MustCall("foo")`.

The other important case is when the programmer has suppressed a warning, for example by annotating a constructor of `Socket` as `@MustCall({})` (as we did with unconnected sockets). In that case, the declared type is `@MustCall("close") Socket`, but the refined type of the object is `@MustCall({})`, so the LUB becomes `@MustCall("close") Socket`. That allows us to handle e.g. calls to `connect()` on unconnected sockets.